src/Bundle/CoreBundle/Form/FieldableFormBuilder.php

```diff
@@ -6,23 +6,47 @@
 use UnitedCMS\CoreBundle\Entity\Fieldable;
 use UnitedCMS\CoreBundle\Entity\FieldableContent;
 use UnitedCMS\CoreBundle\Field\FieldTypeManager;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage;
 
 class FieldableFormBuilder
 {
+    /**
+     * @var TokenStorage $securityTokenStorage
+     */
+    private $securityTokenStorage;
 
     private $formFactory;
     private $fieldTypeManager;
 
-    public function __construct(FormFactory $formFactory, FieldTypeManager $fieldTypeManager)
+    public function __construct(FormFactory $formFactory, FieldTypeManager $fieldTypeManager, TokenStorage $tokenStorage)
     {
         $this->formFactory = $formFactory;
         $this->fieldTypeManager = $fieldTypeManager;
+        $this->securityTokenStorage = $tokenStorage;
+    }
+
+    /**
+     * determine if the current authentication is done via API token or not, disable csrf for API Token
+     *
+     * @param $options
+     *
+     * @return array
+     */
+    public function handleCsrfProtection($options)
```

Comment:
For a cleaner solution, disable csrf protection directly in the form and not in the form builder